File: 프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <string>
#include <queue>
#include <functional>
#include <iostream>
#include <vector>

using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    priority_queue<int,vector<int>,greater<int>> pq(scoville.begin(),scoville.end());
    
    while(pq.top()<K){
        if(pq.size()<=1){
            return -1;
        }
        int f=pq.top();
        pq.pop();
        int s=pq.top();
        pq.pop();
        int t=f+s*2;
        pq.push(t);
        answer++;
    }
    
    return answer;
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for (int s : scoville) {
            pq.offer(s);
        }

        while (pq.peek() < K) {
            if (pq.size() < 2) return -1;
            
            int a = pq.poll();
            int b = pq.poll();
            
            int newScoville = a + (b * 2);
            pq.offer(newScoville);
            answer++;
        }
        
        return answer;
    }
}